src/Database/Query.php

```diff
@@ -1393,6 +1394,27 @@ public function into($table)
     }
 
     /**
+     * Creates an expression that refers to an identifier. Identifiers are used to refer to field names and allow
+     * the SQL compiler to apply quotes or escape the identifier.
+     *
+     * The value is used as is, and you might be required to use aliases or include the table reference in
+     * the identifier. Do not use this method to inject SQL methods or logical statements.
+     *
+     * ### Example
+     *
+     * ```
+     * $query->newExp()->lte('count', $query->identifier('total'));
+     * ```
+     *
+     * @param string $identifier The identifier for an expression
+     * @return ExpressionInterface
+     */
+    public function identifier($identifier)
+    {
+        return new IdentifierExpression($identifier);
```

Comment:
@markstory If the concrete implementation should be kept internal, then a public interface would be nice. Given that things become accessible via a public, overridable API, providing a reliable contract, and allowing userland code to supply custom implementations doesn't seem too unreasonable to me.

Currently the API basically says that `ExpressionInterface` is the lowest common denominator for expressions that represent identifiers, but that's not true.